lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -148,7 +149,7 @@ def bulk_create_models(cls, student, subsection_grades, course_key):
         """
         Saves the subsection grade in a persisted model.
         """
-        subsection_grades = filter(lambda subs_grade: subs_grade._should_persist_per_attempted, subsection_grades)
+        subsection_grades = filter(lambda subs_grade: subs_grade._should_persist_per_attempted(), subsection_grades)
```

Comment:
I think you can rework this whole method body into:
```python
return PersistentSubsectionGrade.bulk_create_grades(
    [subsection_grade._persisted_model_params(student) for subsection_grade in subsection_grades
     if subsection_grade._should_persist_per_attempted()],
    course_key,
)
```

Reply:
...or do it not in a comprehension like that, but you definitely don't have to make two passes over the same list here.